src/item.cpp

```diff
@@ -3024,18 +3026,22 @@ int item::get_encumber() const
     return encumber;
 }
 
-int item::get_layer() const
+int item::get_layer() const {
+	return static_cast<int>( layer() );
+}
+
+layer_level item::layer() const
 {
     if( has_flag("SKINTIGHT") ) {
-        return UNDERWEAR;
+        return layer_level::UNDERWEAR;
```

Comment:
While we'r eat it: what happens if the item has several of these flags (it apparently can only have one layer)?

It seems, the layer should be a separate explicit property of the item type.